### PR TITLE
developer-workflow: Non-functional / Instrumentation section in test-plan

### LINKED ISSUE #147: developer-workflow: добавить Non-functional / Instrumentation секцию в generate-test-plan
## Context

Скилл `plugins/developer-workflow/skills/generate-test-plan/` производит permanent test-plan в `docs/testplans/<slug>-test-plan.md`. Текущий шаблон covers функциональные test cases, edge cases, regression, частично accessibility.

## Problem

Для user-facing / prod-bound feature требуется определить **до реализации**:
- Log events (какие, какие поля, без PII).
- Метрики (counters / gauges / histograms).
- Trace spans (OpenTelemetry conventions).
- Alerts и dashboards.

Сейчас эта работа делается **post-incident** (нет алерта → баг прошёл в prod → добавили метрику реактивно). В test-plan'е нет секции под это, поэтому instrumentation забывается.

## Proposal

Расширить шаблон `test-plan.md` новой секцией `## Non-functional / Instrumentation` с условной активацией.

### Условия активации

**Обязательно заполнить**, если:
- Spec / task риск-флаги включают `user-facing` или `prod-bound`.
- Feature касается observability-hot paths: network calls, payments, background jobs, auth.

**Допускается N/A: reason** для:
- Internal / dev-only.
- Pure refactor без изменения observable behavior.

### Шаблон секции

```markdown
## Non-functional / Instrumentation

### Log events
- Event: `<namespace>.<action>` — когда, поля (без PII)

### Metrics
- Counter: `<name>` — increments on ..., labels ...
- Histogram: `<name>` — observes ..., buckets ...
- Gauge: `<name>` — tracks ...

### Traces
- Span: `<operation>` — entry point, child spans, key attributes
- Parent context: откуда приходит trace-id

### Alerts
- Alert: `<name>` — condition, severity, route (oncall / slack / email)
- Runbook: `runbooks/<slug>.md` (если есть convention)

### Dashboards
- Existing to update: URL/id
- New needed: yes/no, owner
```

### Интеграция с project conventions

- OpenTelemetry stack — использовать OTel naming.
- Custom stack (StatsD, Prometheus, vendor) — использовать проектный namespacing.
- Скилл считывает conventions из проектного CLAUDE.md, не навязывает stack.

## Implementation

- **generate-test-plan/SKILL.md:** добавить в output шаблон новую секцию.
- **generate-test-plan/references/instrumentation-template.md:** под-шаблон для копирования.
- **multiexpert-review test-plan profile** (`plugins/developer-workflow/skills/multiexpert-review/profiles/test-plan.md`): добавить в чеклист проверку «Non-functional / Instrumentation заполнена или явно N/A с reason».
- **acceptance/SKILL.md:** при verification для features с заполненной Instrumentation — проверить, что задекларированные события/метрики действительно генерируются (на running app).

## Acceptance criteria

- [ ] Шаблон `test-plan.md` содержит секцию `## Non-functional / Instrumentation` с Log events, Metrics, Traces, Alerts, Dashboards.
- [ ] Скилл заполняет секцию автоматически из research.md / plan.md или ставит явный N/A с reason.
- [ ] multiexpert-review test-plan profile проверяет секцию.
- [ ] Acceptance для user-facing features проверяет генерацию событий/метрик.
- [ ] Smoke test: test-plan для user-facing feature → секция не пустая.
- [ ] Smoke test: test-plan для internal refactor → секция `N/A: internal refactor, no observable behavior`.

## Non-goals

- Не писать OTel SDK, не выбирать vendor.
- Не добавлять отдельную стадию `observability-plan` в оркестратор — это расширение существующего артефакта.
- Не навязывать formalism проектам без observability stack. N/A с reason — валидный ответ.

## Dependencies

- Блокеры: нет.
- Связано: может переиспользоваться в bugfix-flow для bugs, где root cause = отсутствие телеметрии.

## Effort / Risk

- Effort: S (шаблон + чеклист в profile + acceptance шаг).
- Risk: пользователи могут писать N/A formально. Митигация: чёткие условия «обязательно для user-facing / prod-bound».

### PR DESCRIPTION
**Pull request by kirich1409**
## Summary

Closes #147.

- `generate-test-plan/references/format-templates.md` adds a `## Non-functional / Instrumentation` template — Log events / Metrics / Traces / Alerts / Dashboards. Mandatory for `user-facing` / `prod-bound` features and observability hot-paths (network, payments, background jobs, auth, data migrations). Explicit `N/A: <reason>` allowed for internal / dev-only / pure-refactor work.
- `generate-test-plan/SKILL.md` Field Definitions describe the section, activation rule, and downstream consumers. Naming and stack come from the project's `CLAUDE.md` — the skill records what is already in use, asks one question only when the project has no convention.
- `multiexpert-review/profiles/test-plan.md` gains item **(g) Instrumentation declared** — major severity (WARN, not blocking), consistent with items (d) / (e). Severity mapping updated.
- `acceptance/SKILL.md` adds an Instrumentation verification note: when the test plan declares events / metrics, acceptance verifies they actually fire against the running app; mismatch is a P1 finding routed through the standard FAILED → Implement loop.

## Acceptance criteria mapping (from #147)

- [x] Test-plan template contains the section with all five subsections.
- [x] Skill fills the section automatically from `research.md` / `plan.md`, or marks an explicit `N/A: <reason>`.
- [x] `multiexpert-review` test-plan profile checks the section.
- [x] `acceptance` for user-facing features verifies declared events / metrics on the running app.
- [ ] Smoke test: a test plan for a user-facing feature emits a non-empty section — verifiable on the next real run.
- [ ] Smoke test: a test plan for an internal refactor emits `N/A: internal refactor, no observable behavior` — verifiable on the next real run.

## Notes for review

- **Letter (g) on the multiexpert checklist**, not (f). Letter (f) is in flight in PR #181 (#153 — TC type field). Both PRs touch `multiexpert-review/profiles/test-plan.md`; using `(g)` here keeps the change orthogonal at the line level. The `severity_mapping` references items by id, not by sequence — order does not matter.
- **No new orchestrator stage.** Per the parent strategy (#151), this is an extension of an existing artifact, not a new stage.

## Out of scope

- Writing OpenTelemetry SDK or vendor selection (#147 Non-goals).
- Adding a separate `observability-plan` orchestrator stage (#147 Non-goals — the min-bar #148/#175 redirects this kind of proposal to artifact extension).
- Forcing observability stack on projects without one (#147 Non-goals — `N/A: <reason>` is a valid answer).

## Dependencies

- Coexists with PR #181 (#153 type field). If PR #181 merges first, this PR's `(g)` simply lands after `(f)`. If this PR merges first, PR #181's `(f)` lands without conflict.

## Test plan

- [x] `bash scripts/validate.sh` — green.
- [ ] Manual: trigger `generate-test-plan` for a `user-facing` task — confirm the section is generated with concrete events / metrics.
- [ ] Manual: trigger `generate-test-plan` for an internal refactor — confirm the section emits `N/A: <reason>`.
- [ ] Manual: confirm `multiexpert-review` flags a missing / blank instrumentation section as WARN.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/developer-workflow/skills/acceptance/SKILL.md [hunk 1]
```diff
@@ -90,6 +90,14 @@ consumers (including the `surfaces` invariant guards) live in
 [`references/source-branches.md`](references/source-branches.md). Record the selected
 branch as `test_plan_source` in the receipt.
 
+**Instrumentation verification.** When the test plan ends with a non-empty
+`## Non-functional / Instrumentation` section (Log events / Metrics / Traces / Alerts /
+Dashboards — see [`generate-test-plan` Field Definitions](../generate-test-plan/SKILL.md#non-functional--instrumentation-mandatory-for-user-facing--prod-bound)),
+acceptance verifies, against the running app, that each declared event / metric / span
+fires when its tested behaviour runs. Mismatch (declared but not emitted, or emitted with
+wrong fields) becomes a P1 acceptance finding routed through the standard FAILED → Implement
+loop. An explicit `N/A: <reason>` in the test-plan section skips this check.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/acceptance/SKILL.md]: Resolved in 767d684: trigger now reads "section that exists and is not `N/A: <reason>`" — `N/A:` skips the check, only filled sections (Log events / Metrics / Traces / Alerts / Dashboards or any subset) trigger verification. The Alerts/Dashboards scope point you raised separately (these are configuration, not runtime emissions) is real but lives outside the wording ambiguity Copilot flagged here; tracking it as follow-up rather than expanding this PR. The current text only covers what was asked.

File: plugins/developer-workflow/skills/generate-test-plan/SKILL.md [hunk 2]
```diff
@@ -195,6 +195,19 @@ markdown), the phase-segmentation worked example, and the rules for when each va
 | Code path | backtick-wrapped path with line | `src/auth/LoginViewModel.kt:87` |
 | Inferred | `[inferred from code]` | Behavior derived from code with no spec backing |
 
+### Non-functional / Instrumentation (mandatory for user-facing / prod-bound)
+
+Every plan ends with a `## Non-functional / Instrumentation` section that declares observability **before** implementation, not after the first incident. Required when the spec / task is tagged `user-facing` or `prod-bound`, or when the feature touches an observability hot-path: network calls, payments, background jobs, auth, data migrations.
+
+`N/A: <reason>` (one line) is allowed for internal / developer-only tooling and for pure refactors with no change to observable behaviour. Never delete the heading.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/generate-test-plan/SKILL.md]: Resolved in 767d684: changed `behaviour` -> `behavior` to match the rest of the file (US spelling).

File: plugins/developer-workflow/skills/generate-test-plan/SKILL.md [hunk 3]
```diff
@@ -195,6 +195,19 @@ markdown), the phase-segmentation worked example, and the rules for when each va
...
+
+The section covers five subsections — Log events / Metrics / Traces / Alerts / Dashboards (full template in [`references/format-templates.md`](references/format-templates.md#non-functional--instrumentation)). The skill reads naming and stack conventions (OpenTelemetry, Prometheus, StatsD, vendor-specific) from the project's `CLAUDE.md` and reuses them; it does not prescribe a stack. If the project has no convention, the skill asks one question and records the answer.
+
+Downstream stages consume this section:
+
+- `multiexpert-review` test-plan profile checks the section is filled or carries an explicit `N/A: <reason>`.
+- `acceptance` verifies, against the running app, that declared events / metrics actually fire when the tested behaviour runs.
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/generate-test-plan/SKILL.md]: Resolved in 767d684: changed `behaviour` -> `behavior` to match the rest of the file (US spelling).

File: plugins/developer-workflow/skills/multiexpert-review/profiles/test-plan.md [hunk 4]
```diff
@@ -44,27 +44,28 @@ receipt:
...
 
 ## Verdict policy
 
 | Verdict | Trigger | Exit condition |
 |---------|---------|----------------|
 | **FAIL** (blocker) | Any of items **(a)**, **(b)**, **(c)** is violated | Plan MUST be revised. Engine drives the revise-loop up to 3 cycles. After 3 cycles still FAIL → escalate to user. Pipeline is blocked. |
-| **WARN** (non-blocking) | Items (a)–(c) all satisfied, but **(d)** or **(e)** is violated | Pipeline continues. Engine records `review_verdict: WARN` in the receipt with the explicit list of violated items. No revise-loop required. |
-| **PASS** (clean) | All five items satisfied | Pipeline continues unconditionally. Engine records `review_verdict: PASS` in the receipt. |
+| **WARN** (non-blocking) | Items (a)–(c) all satisfied, but **(d)**, **(e)**, or **(g)** is violated | Pipeline continues. Engine records `review_verdict: WARN` in the receipt with the explicit list of violated items. No revise-loop required. |
+| **PASS** (clean) | Every item satisfied | Pipeline continues unconditionally. Engine records `review_verdict: PASS` in the receipt. |
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/multiexpert-review/profiles/test-plan.md:62]: Resolved in 767d684: profile description now says `6-item checklist (a, b, c, d, e, g)`, and the Receipt integration WARN bullet enumerates violated items from `(d), (e), (g)`. The Verdict policy table at L61 already references `(d)`, `(e)`, or `(g)` — no change needed there.
